src/coins.cpp

```diff
@@ -299,12 +301,48 @@ void CCoinsViewCache::BringBestAnchorIntoCache(
     assert(GetSaplingAnchorAt(currentRoot, tree));
 }
 
+
+std::string ToStr(uint32_t i) {
+    return std::to_string(i);
+}
+
+std::string ToStr(const HistoryNode& node) {
+    return HexStr(std::begin(node.bytes), std::end(node.bytes));
+}
+
+std::string ToStr(const HistoryEntry& entry) {
+    return HexStr(std::begin(entry.bytes), std::end(entry.bytes));
+}
+
+template<typename T>
+std::string ToStr(T it, const T itend) {
+    std::stringstream str;
+    str << "{";
+    if (it != itend) {
+        str << ToStr(*it);
+        it++;
+    }
+    while (it != itend) {
+        str << ", " << ToStr(*it);
+        it++;
+    }
+    str << "}";
+    return str.str();
+}
```

Comment:
I considered moving these to `utilstrencodings.{h,cpp}`, but decided it was overkill for this debugging PR. That might be a useful refactoring for later.